dali/operators/input/video_input_test.cc

```diff
@@ -54,5 +60,291 @@ TEST(VideoInputTest, DetermineBatchOutlineTest) {
+    FeedExternalInput(h, LoadTestFile(test_files_[test_file_idx].file_name),
+                      test_files_[test_file_idx].data_id);
+
+    // Determine, how many iterations the input test file should produce.
+    auto [num_full_batches, num_full_sequences, frames_in_last_sequence] =
+            detail::DetermineBatchOutline(
+                    test_files_[test_file_idx].n_frames, frames_per_sequence_, batch_size_);
+    auto num_iterations_per_input =
+            num_full_batches + (num_full_sequences > 0 || frames_in_last_sequence > 0 ? 1 : 0);
+
+    // Run the pipeline to test it.
+    for (int i = 0;
+         // `-1`, since the last iteration will carry different result,
+         // so it will be checked outside the loop.
+         i < num_iterations_per_input - 1;
+         i++) {
+      daliRun(h);
+      daliOutput(h);
+
+      DoesOperatorTraceExist(h, i, test_file_idx);
+      IsOperatorTraceCorrect(h, i, test_file_idx);
+    }
+    /*
+     * The last iteration of the pipeline shall carry a different result.
+     * Since this function tests a single file, after the last iteration there shouldn't
+     * be a "next_output_data_id" trace available.
+     */
+    daliRun(h);
+    daliOutput(h);
+    EXPECT_EQ(daliHasOperatorTrace(h, video_input_name_.c_str(), trace_name_.c_str()), 0);
+  }
+
+
+  void CreateAndSerializePipeline() {
+    auto pipeline = std::make_unique<Pipeline>(batch_size_, num_threads_, device_id_);
+    pipeline->AddOperator(OpSpec("experimental__inputs__Video")
+                                  .AddArg("sequence_length", frames_per_sequence_)
+                                  .AddArg("device", "cpu")
+                                  .AddArg("name", video_input_name_)
+                                  .AddOutput(video_input_name_, "cpu"),
+                          video_input_name_);
+
+    std::vector<std::pair<std::string, std::string>> outputs = {
+            {video_input_name_, "cpu"},
+    };
+
+    pipeline->SetOutputDescs(outputs);
+
+    serialized_pipeline_ = pipeline->SerializeToProtobuf();
+  }
+
+
+  void FeedExternalInput(daliPipelineHandle *h, const std::string &encoded_video,
+                         const std::string &data_id) {
+    daliSetExternalInputBatchSize(h, video_input_name_.c_str(), 1);
+    if (!data_id.empty()) {
+      daliSetExternalInputDataId(h, video_input_name_.c_str(), data_id.data());
+    }
+    int64_t shapes[] = {static_cast<int64_t>(encoded_video.length())};
+    daliSetExternalInput(h, video_input_name_.c_str(), device_type_t::CPU, encoded_video.data(),
+                         dali_data_type_t::DALI_UINT8, shapes, 1, nullptr, DALI_ext_force_copy);
+  }
+
+
+  /**
+   * Check, if the "next_output_data_id" trace exists, provided it should.
+   */
+  void DoesOperatorTraceExist(daliPipelineHandle *h, int iteration_idx, int test_file_idx) {
+    bool has_data_id = daliHasOperatorTrace(h, video_input_name_.c_str(), trace_name_.c_str());
```

Comment:
This API design is inefficient - daliGetOperatorTrace returns `const char *`, so there's an obvious NULL value which would indicate the absence of the trace. No need for two separate calls.
Also, while we're at it, I'd recommend changing the API to
`daliGetOperatorTrace(daliPipelineHandle pipe_handle, const char **value, const char *operator_name,
                     const char *trace_name);`
This way we'll be able to accommodate the error code in the future.

Reply:
Agree. Let's do it in a followup PR, since change in this API will stipulate changes in API tests. Will scope more files, not related to this PR.